.github/workflows/custard-run.yaml

```diff
@@ -0,0 +1,166 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: (experimental) Custard run
+
+on:
+  push:
+    branches:
+      - main
+  workflow_run:
+    workflows:
+      - Custard CI # .github/workflows/custard-ci.yaml
+    types:
+      - in_progress
+  workflow_dispatch:
+    inputs:
+      run-all:
+        description: Run all tests
+        type: boolean
+        default: false
+      paths:
+        description: Comma separated packages to test
+        type: string
+      ref:
+        description: Branch, tag, or commit SHA to run tests on
+        type: string
+        default: main
+  # schedule:
```

Comment:
Will we ever want to schedule tests on forked runs? If not we could remove this comment.

Reply:
The schedule runs from main. This would basically enable nightly tests. I haven't tested it yet so that's why it's commented out, but we eventually want to enable schedule for nightlies. There would be a special case handling on affected to run on everything if it's a schedule (rather than diffs).